data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/PipelineConnector.java

```diff
@@ -54,6 +56,21 @@ public void stop() {
     public void output(final Collection<T> records) {
         if (buffer != null && !isStopRequested.get()) {
             for (T record : records) {
+		if(record.getData() instanceof Event) {
```

Comment:
It could be good to add some unit tests for this in `PipelineConnectorTest`.

Reply:
Will do

Reply:
Added test case and updated diffs